npm_and_yarn/lib/dependabot/npm_and_yarn/yarn_error_handler.rb

```diff
@@ -0,0 +1,260 @@
+      "YN0071" => { message: "NM can't install external soft link", error_class: Dependabot::DependabotError },
+      "YN0072" => { message: "NM preserve symlinks required", error_class: Dependabot::DependabotError },
+      "YN0074" => { message: "NM hardlinks mode downgraded", error_class: Dependabot::DependabotError },
+      "YN0075" => { message: "Prolog instantiation error", error_class: Dependabot::DependabotError },
+      "YN0076" => { message: "Incompatible architecture", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0077" => { message: "Ghost architecture", error_class: Dependabot::DependabotError },
+      "YN0078" => { message: "Resolution mismatch", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0080" => { message: "Network disabled", error_class: Dependabot::DependabotError },
+      "YN0081" => { message: "Network unsafe HTTP", error_class: Dependabot::DependabotError },
+      "YN0082" => { message: "Resolution failed", error_class: Dependabot::DependencyFileNotResolvable },
+      "YN0083" => { message: "Automerge git error", error_class: DependabotError },
+      "YN0085" => { message: "Updated resolution record", error_class: Dependabot::DependabotError },
+      "YN0086" => { message: "Explain peer dependencies CTA", error_class: Dependabot::DependabotError },
+      "YN0087" => { message: "Migration success", error_class: Dependabot::DependabotError },
+      "YN0088" => { message: "Version notice", error_class: Dependabot::DependabotError },
+      "YN0089" => { message: "Tips notice", error_class: Dependabot::DependabotError },
+      "YN0090" => { message: "Offline mode enabled", error_class: Dependabot::DependabotError }
+    }.freeze
+
+    # Used to check if package manager registry is public npm registry
+    NPM_REGISTERY = "registry.npmjs.org"
+
+    # Used to check if url is http or https
+    HTTP_CHECK_REGEX = %r{https?://}
+
+    # Error message when a package.json name include invalid characters
+    INVALID_NAME_IN_PACKAGE_JSON = "Name contains illegal characters"
+
+    # lambda function for finding missing package
+    PACKAGE_MISSING_REGEX = /(ENOBUFS|ETIMEDOUT|The registry may be down)/
+
+    # Used to check if error message contains timeout fetching package
+    TIMEOUT_FETCHING_PACKAGE_REGEX = %r{(?<url>.+)/(?<package>[^/]+): ETIMEDOUT}
+
+    # Used to check if git dependency is unreachable
+    UNREACHABLE_GIT_CHECK_REGEX = /ls-remote --tags --heads (?<url>.*)/
+
+    # Used to check if yarn workspace is enabled in non-private workspace
+    ONLY_PRIVATE_WORKSPACE_TEXT = "Workspaces can only be enabled in priva"
+
+    # Used to check if sub-dependency is local path
+    SUB_DEP_LOCAL_PATH_TEXT = "refers to a non-existing file"
+    # Error message when a package.json doesn't include a name or version
+    INVALID_PACKAGE_REGEX = /Can't add "(?<package_req>.*)": invalid/
```

Comment:
Seems like all of these constants belong to the error handler class. No need to expose them at this level. If we do need to add them to this scope for some reason, it would be better to place them in the `lib/dependabot/npm_and_yarn.rb` file, as that would be the idiomatic place to define `NpmAndYarn` and where one would expect things on that module to be defined

Reply:
I moved the constants. Is this what you were mentioning or moving all error handling process as well? I am really wondering if file structure is ok or not now.

Reply:
Also, after doing some investigation most of yarn error codes removed from the map and each yarn error code mapped to a newly created/defined error code.